### PR TITLE
MAGECLOUD-4458: De-compose All Patches from ECE-Tools

### PR DESCRIPTION
**Pull request by shiftedreality**
### Fixed Issues (if relevant)

1. https://magento2.atlassian.net/browse/MAGECLOUD-4458

### Manual testing scenarios

1. Edit `composer.json`:
```json
    "repositories": {
        "repo": {
            "type": "composer",
            "url": "https://repo.magento.com"
        },
        "ece-tools": {
            "type": "vcs",
            "url": "git@github.com:magento/ece-tools.git"
        },
        "mcp": {
            "type": "vcs",
            "url": "git@github.com:magento/magento-cloud-patches.git"
        }
    },
```

```json
    "require": {
        "magento/magento-cloud-metapackage": ">=2.3.0 <2.3.1",
        "magento/ece-tools": "dev-MAGECLOUD-4458 as 2002.0.99",
        "magento/magento-cloud-patches": "1.0.x-dev as 1.0.0"
    },
```
2. Perform acceptance tests

### Related Pull Requests
* https://github.com/magento/magento-cloud-patches/pull/2

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -29,14 +29,15 @@
     "symfony/process": "~2.1||~4.1.0",
     "symfony/yaml": "^3.3||^4.0",
     "twig/twig": "^1.0||^2.0",
-    "magento/magento-cloud-components": "^1.0.1"
+    "magento/magento-cloud-components": "^1.0.1",
+    "magento/magento-cloud-patches": "^1.0"
```
Comment (NadiyaS) [inline composer.json]: why for `cloud-components` we allow minor releases and for `patches` only patch releases? 
I am not against this, I just want to know the reason of this decision.

Comment (shiftedreality) [reply composer.json]: This is the same as `^1.0.0`, so until `1.0.1` is released it does not make sense to add patch constraint part

Comment (shiftedreality) [reply composer.json]: https://semver.mwl.be/#!?package=madewithlove%2Felasticsearcher&version=%5E0.5&minimum-stability=stable vs https://semver.mwl.be/#!?package=madewithlove%2Felasticsearcher&version=%5E0.5.0&minimum-stability=stable

Comment (NadiyaS) [reply composer.json]: I mixed up with `^0.1` which is equal to `>=0.1.0 <0.2.0`

Comment (NadiyaS) [reply composer.json]: Btw, I did not get what you wanted to show with those links. They do not show the real rule from this constraint

Comment (shiftedreality) [reply composer.json]: I just showed there is no difference betweeen `^1.0` and `^1.0.0` at this point

**Review by NadiyaS — DISMISSED** [review]

**Comment by andriyShevtsov** [comment]
As far as related  PR was merged https://github.com/magento/magento-cloud-patches/pull/2 I believe we can remove on-hold label  from this

File: codeception.dist.yml [hunk 2]
```diff
@@ -31,7 +31,7 @@ modules:
         - /var/www/magento/pub/media
         - /var/www/magento/var
         - /var/www/magento/app/etc
-      printOutput: false
+      printOutput: true
```
Comment (oshmyheliuk) [inline codeception.dist.yml]: Is it a good idea to change this value? Logs in Travis can be very huge.

File: src/Patch/Manager.php [hunk 3]
```diff
@@ -18,158 +18,88 @@
...
-    private function copyStaticFile()
+    public function apply()
     {
         $magentoRoot = $this->directoryList->getMagentoRoot();
 
         if (!$this->file->isExists($magentoRoot . '/pub/static.php')) {
-            $this->logger->notice('File static.php was not found.');
-
-            return;
+            $this->logger->notice('File static.php was not found');
```
Comment (oshmyheliuk) [inline src/Patch/Manager.php:77]: I think we should move this functionality from here, this is not related to patch applying.

File: codeception.dist.yml (hunk 2 above)
Comment (shiftedreality) [reply codeception.dist.yml]: Forgotten debug stmt, removed

File: src/Patch/Manager.php (hunk 3 above)
Comment (shiftedreality) [reply src/Patch/Manager.php:77]: I'll keep in this 2002.0 branch and refactor in PR to `develop` as it does not give any profit

Comment (oshmyheliuk) [reply src/Patch/Manager.php:77]: ok, agree

**Review by oshmyheliuk — APPROVED** [review]

**Review by NadiyaS — APPROVED** [review]

**Comment by andriyShevtsov** [comment]
QA approved 
Tested in scope of:
- https://github.com/magento/magento-cloud-patches/pull/3
- https://github.com/magento/magento-cloud-patches/pull/2

**Comment by ghost** [comment]
Hi @shiftedreality, thank you for your contribution! </br>Please, complete [Contribution Survey](https://opensource.magento.com/survey/contribution/magento+ece-tools+630), it will take less than a minute. </br>Your feedback will help us to improve contribution process.
